docs/dg/dev/miscellaneous-guides/strategy-resolver.md

```diff
@@ -0,0 +1,163 @@
+---
+title: Strategy Resolvers
+description: This document
+last_updated: Feb 6, 2025
+template: howto-guide-template
+redirect_from:
+- /docs/dg/dev/architecture/programming-concepts.html
+
+---
+
+Spryker introduces context-based dependency resolution, such as plugin-stack, using a strategy resolver to handle complex workflows that stretch across multiple modules. This enhancement allows for defining multiple dependencies and selecting the appropriate one dynamically at runtime by the business logic.
```

Comment:
What's an appropriate dependency?

Reply:
So, 1 plugin stack that uses strategy resolving, has multiple context. Each context is basically a separate individual plugin stack (with many plugins). When you seelct a context, you actually say which plugin stack to execute.